sql: add empty pg_event_trigger

OwenQian opened the pull request:
The event trigger table is currently empty, but will be populated in the future to improve compatibility with postgres.

Release note: None.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47740)
<!-- Reviewable:end -->

Commit:
sql: add empty pg_event_trigger

The event trigger table is currently empty, but will be populated in the future
to improve compatibility with postgres.

Release note: None.

Fixes #47442.

Rewrite logic test

nvb reviewed (commented):
Nice! I think this is close. Just a few comments and we also need to get CI passing.

> Reviewed 5 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47740#-:-M5SBx8D2deaIohHAn5d:b-x7xt4v)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @OwenQian)

___
*[pkg/sql/pg_catalog.go, line 214 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47740#-M5SB_ex2TiBHO2OaT4u:-M5SB_ex2TiBHO2OaT4v:b-jrfbv) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8898e2578882382f46bdfdbfd0876bf7fde088e/pkg/sql/pg_catalog.go#L214)):*
> ```Go
> 		sqlbase.PgCatalogSharedDescriptionTableID:   pgCatalogSharedDescriptionTable,
> 		sqlbase.PgCatalogEnumTableID:                pgCatalogEnumTable,
> 		sqlbase.pgCatalogEventTriggerTableID:        pgCatalogEventTriggerTable,
> ```

Hm, does this compile?
___
*[pkg/sql/pg_catalog.go, line 1359 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47740#-M5SAhpk534QL5ci660p:-M5SAhpk534QL5ci660q:b-r8dnzz) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8898e2578882382f46bdfdbfd0876bf7fde088e/pkg/sql/pg_catalog.go#L1359)):*
> ```Go
> 
> var pgCatalogEventTriggerTable = virtualSchemaTable{
> 	schema: `
> ```

Give this a comment like `pgCatalogExtensionTable.comment`.
___
*[pkg/sql/sqlbase/constants.go, line 129 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47740#-M5SBf2lAW5XdPDdg55z:-M5SBf2lAW5XdPDdg56-:b-502e4s) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8898e2578882382f46bdfdbfd0876bf7fde088e/pkg/sql/sqlbase/constants.go#L129)):*
> ```Go
> 	PgCatalogSharedDescriptionTableID
> 	PgCatalogEnumTableID
> 	pgCatalogEventTriggerTableID
> ```

I'm surprised this doesn't need to be exported. Doesn't it?


<!-- Sent from Reviewable.io -->

OwenQian reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47740)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @nvanbenschoten)

___
*[pkg/sql/pg_catalog.go, line 214 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47740#-M5SB_ex2TiBHO2OaT4u:-M5TPPpuACUykoyw40g7:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8898e2578882382f46bdfdbfd0876bf7fde088e/pkg/sql/pg_catalog.go#L214)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Hm, does this compile?
</blockquote></details>

Done.
___
*[pkg/sql/pg_catalog.go, line 1359 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47740#-M5SAhpk534QL5ci660p:-M5TPLkN8HO4jJcYryoA:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8898e2578882382f46bdfdbfd0876bf7fde088e/pkg/sql/pg_catalog.go#L1359)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Give this a comment like `pgCatalogExtensionTable.comment`.
</blockquote></details>

Done.
___
*[pkg/sql/sqlbase/constants.go, line 129 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47740#-M5SBf2lAW5XdPDdg55z:-M5TPN-pDGe9SlwKIPlh:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8898e2578882382f46bdfdbfd0876bf7fde088e/pkg/sql/sqlbase/constants.go#L129)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

I'm surprised this doesn't need to be exported. Doesn't it?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

nvb reviewed (commented):
Getting closer, but this still isn't passing tests because of a syntax error. I'd suggest running `make test PKG=./pkg/sql/logictest` and confirming that that passes.

> Reviewed 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47740#-:-M5TSC_HAUfMwjeDEO7b:b-xo4c4v)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @OwenQian)

___
*[pkg/sql/pg_catalog.go, line 1359 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47740#-M5SAhpk534QL5ci660p:-M5TS1bz8CfE1sTGIeru:b-kpydw0) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8898e2578882382f46bdfdbfd0876bf7fde088e/pkg/sql/pg_catalog.go#L1359)):*
<details><summary><i>Previously, OwenQian (Owen Qian) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Also move it above the `schema` field to keep the ordering consistent.
___
*[pkg/sql/pg_catalog.go, line 1360 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47740#-M5TSV-m3sAXwHMQBKrE:-M5TSV-m3sAXwHMQBKrF:by5hgey) ([raw file](https://github.com/cockroachdb/cockroach/blob/e42856055600e847df1e693f523eb1df24299cfa/pkg/sql/pg_catalog.go#L1360)):*
> ```Go
> var pgCatalogEventTriggerTable = virtualSchemaTable{
> 	schema: `
> CREATE TABLE pg_catalog.pg_event_trigger {
> ```

These should be parentheses, not brackets.


<!-- Sent from Reviewable.io -->

nvanbenschoten commented:
@OwenQian do you mind rebasing this on master and pushing again?

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> let's squash these commits down to one and then merge!

> Reviewed 1 of 1 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47740#-:-M5ZkyMd3HsnLSey1uoe:bqql1ag)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

OwenQian commented:
bors r+
